spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -3064,6 +3088,40 @@ def visible_blocks(self):
     def is_editor(self):
         return True
 
+    def popup_docstring(self, prev_text, prev_pos):
+        """Show the menu for generating docstring"""
+        line_text = self.textCursor().block().text()
+        if line_text != prev_text:
+            return
+
+        if prev_pos != self.textCursor().position():
+            return
+
+        writer = self.writer_docstring
+        if writer.get_function_definition_from_below_last_line():
+            point = self.cursorRect().bottomRight()
+            point = self.calculate_real_position(point)
+            point = self.mapToGlobal(point)
+
+            self.menu_docstring = QMenuOnlyForEnter(self)
+            self.docstring_action = create_action(
+                self, _("Write docstring"), icon=ima.icon('TextFileIcon'),
+                triggered=writer.write_docstring)
+            self.menu_docstring.addAction(self.docstring_action)
+            self.menu_docstring.setActiveAction(self.docstring_action)
+            self.menu_docstring.popup(point)
+
+    def delayed_popup_docstring(self):
+        """Show context menu for docstring.
+        This method is called after typing '''. After typing ''', This function
+        wait 300ms. If there was no input for 300ms, show the context menu.

```

Comment:
Put the summary line on its own line (below the ``"""``) and make sure to leave a blank line between it and the extended description, to follow PEP 257.

Reply:
Also, lowercase ``T`` in ``this function`` since it doesn't begin a sentence, and should be ``waits 300 ms``.